### ISSUE TITLE
Type converter signature should be changed (or extended)

### ISSUE THREAD

**Issue by jenisys** [r1chardj0n3s/parse#19]
The type converter function signature should be changed to:

``` python
def type_converter(text, match=None, match_start=0):
    # -- NEW: match_start : int = 0, 
    # refers to the first group in the match object for a field where the converter is used.
    pass
```

This change would allow to provide a generic Parser without type knowledge.
In addition, user-defined types should also provide this signature. Currently, only the first parameter is supported there which prevents to use complexes type converter cases.

Due to backward compatibility reasons, the old user-defined signature should be supported, too, at least for some time.

**Comment by moi90** [comment]
Yes, it would be great if one were able to access the re.Match object to extract individual groups, for example!
